pygpu/gpuarray.pyx

```diff
@@ -1412,7 +1412,10 @@ cdef GpuArray pygpu_index(GpuArray a, const ssize_t *starts,
                           const ssize_t *stops, const ssize_t *steps):
     cdef GpuArray res
     res = new_GpuArray(type(a), a.context, a.base)
-    array_index(res, a, starts, stops, steps)
+    try:
+        array_index(res, a, starts, stops, steps)
+    except ValueError, e:
```

Comment:
Why this is needed? Can you add a test to make sure this don't get broken again?